### ISSUE TITLE
between function

### ISSUE THREAD

**Issue by jorainer** [rformassspectrometry/MsCoreUtils#105]
`MsCoreUtils` has a `between` function implemented, but there is also another (quite powerful) function available in `dplyr`:

```r
library(microbenchmark)
library(MsCoreUtils)
library(dplyr)

set.seed(123)
rt <- rnorm(n = 100, mean = 50)
microbenchmark(MsCoreUtils::between(rt, c(50, 51)),
               dplyr::between(rt, 50, 51))
Unit: microseconds
                                expr   min     lq    mean median     uq    max
 MsCoreUtils::between(rt, c(50, 51)) 1.359 1.6120 2.36456 2.1760 2.7255 14.798
          dplyr::between(rt, 50, 51) 1.043 1.1815 1.69199 1.4325 1.7045 16.495
 neval cld
   100   b
   100  a 

rt <- rnorm(n = 1000, mean = 50)
microbenchmark(MsCoreUtils::between(rt, c(50, 51)),
               dplyr::between(rt, 50, 51))
Unit: microseconds
                                expr   min     lq    mean median     uq    max
 MsCoreUtils::between(rt, c(50, 51)) 4.505 4.7925 5.16017 4.9355 5.1095 19.458
          dplyr::between(rt, 50, 51) 2.545 2.7270 2.98997 2.8225 2.9605 12.741
 neval cld
   100   b
   100  a 
```

should we maybe replace the existing `between` function with the one from `dplyr` - or remove the `between` function from `MsCoreUtils` (which might however cause some packages that already use it to fail).

**Comment by jorainer** [comment]
Great @sgibb ! Yes, please go ahead and make a PR with the C implementation of `between` - and for the interface, I would stick to our `between(x, range)` which to me is the more natural way.

**Comment by sgibb** [comment]
Closed by #106;

**Comment by lgatto** [comment]
I'm late to the party, but I would also prefer not to depend on `dplyr` as it has quite a few dependencies itself and it (as are the dependencies) updated regularly, which could lead to additional maintenance burden for us.
